deployment/cdk/opensearch-service-migration/lib/opensearch-domain-stack.ts

```diff
@@ -117,6 +119,18 @@ export class OpenSearchDomainStack extends Stack {
     }
   }
 
+  generateTargetClusterYaml(domain: Domain, adminUserName: string | undefined) {
+    let targetCluster = new ClusterYaml()
+    targetCluster.endpoint = `https://${domain.domainEndpoint}:443`;
+    if (adminUserName && domain.masterUserPassword) {
+      // TODO: need to actually deal with the password as a secret (pass the arn)
+      targetCluster.basic_auth = { user: adminUserName, passsword: domain.masterUserPassword }
+    } else {
+      targetCluster.no_auth = ''
```

Comment:
why is no_auth a string and not a boolean?  Or is it really just null vs non-null?

Reply:
Yeah, it's just testing what's present -- so it'll be:
```yaml
target_cluster:
  endpoint: xyz
  no_auth:
```
vs
```
target_cluster:
  endpoint: xyz
  basic_auth:
    user: ...
```